### ISSUE TITLE
Checkpointer should support checkpointing pod labels

### ISSUE THREAD

**Issue by aaronlevy** [kubernetes-incubator/bootkube#378]
They are currently stripped when sanitizing the pod spec - but these are needed for checkpointed pods to satisfy service endpoints.

**Comment by jamiehannaford** [comment]
Just took a look at this and saw your [comment](https://github.com/kubernetes-incubator/bootkube/blob/master/cmd/checkpoint/main.go#L300-L302). What do you mean by "be adopted by higher-level parent"? As in, the list of labels for the static pod would need to exclude `checkpointer.alpha.coreos.com/checkpoint`?

**Comment by aaronlevy** [comment]
The higher-level parent being an object that uses label selectors to determine what pods it is managing (e.g. replicaSet).

We may need to add an entropy label to the checkpointed pod such that it is not attempted to be directly managed by the parent object.

For example:

- A pod is created from a replicaSet, then we checkpoint it.
- The checkpoint is started as a static pod, then a "mirror pod" is created in the api (read-only)
- The replicaSet sees this mirror pod with the correct label selectors and thinks it should be managing it.
- Maybe based on the # of desired replicas, the replicaSet tries to delete the mirror pod (but fails because it is read-only).

This is going from memory of behavior, but it could have change. I tried to find an upstream issue that mentioned this (controllers trying to make changes to mirror pods but failing). So it may have been resolved by now - but didn't immediately find the issue.
